diff --git a/src/fastgltf.cpp b/src/fastgltf.cpp
index 9244fe1..f968f59 100644
--- a/src/fastgltf.cpp
+++ b/src/fastgltf.cpp
@@ -1487,7 +1487,8 @@ fg::Error fg::Parser::parseBuffers(simdjson::dom::array& buffers, Asset& asset)
         // file. Otherwise, data must be specified in the "uri" field.
         std::string_view uriString;
         if (bufferObject["uri"].get_string().get(uriString) == SUCCESS) {
-			URIView uriView(uriString);
+            fg::URI uri(uriString);
+            fg::URIView uriView(uri.string());
 
             if (!uriView.valid()) {
                 return Error::InvalidURI;
@@ -1821,7 +1822,8 @@ fg::Error fg::Parser::parseImages(simdjson::dom::array& images, Asset& asset) {
                 return Error::InvalidGltf;
             }
 
-            URIView uriView(uriString);
+            fg::URI uri(uriString);
+            fg::URIView uriView(uri.string());
             if (!uriView.valid()) {
                 return Error::InvalidURI;
             }
